openmeter/entitlement/metered/grant_owner_adapter.go

```diff
@@ -201,14 +209,18 @@ func (e *entitlementGrantOwner) updateEntitlementUsagePeriod(ctx context.Context
 		})
 }
 
-// FIXME: this is a terrible hack using select for udpate...
-func (e *entitlementGrantOwner) LockOwnerForTx(ctx context.Context, tx *entutils.TxDriver, owner grant.NamespacedOwner) error {
-	return e.entitlementRepo.WithTx(ctx, tx).LockEntitlementForTx(ctx, owner.NamespacedID())
+func (e *entitlementGrantOwner) LockOwnerForTx(ctx context.Context, owner grant.NamespacedOwner) error {
+	// If we're not in a transaction this method has to fail
+	_, err := transaction.GetDriverFromContext(ctx)
+	if err != nil {
+		return fmt.Errorf("lock owner for tx must be called in a transaction: %w", err)
+	}
+	return e.locker.LockForCtx(ctx, owner.NamespacedID().ID)
```

Comment:
Why don't we lock the owner's record instead of relying on a separate lock table?

Reply:
Because we might be updating the entitlement entity itself.

Reply:
AFAIK you can still update while having a select for update lock on the row.

Reply:
```
postgres=# CREATE TABLE test (
postgres(#   id numeric,
postgres(#   value text);
CREATE TABLE

postgres=# insert into test values (1, 'a');
INSERT 0 1
postgres=# insert into test values (2, 'b');
INSERT 0 1

postgres=# BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
BEGIN

postgres=*# SELECT  * FROM test where id = 1 FOR UPDATE;
 id | value
----+-------
  1 | a
(1 row)

postgres=*# update test set value = 'xxx' where id = 1;
UPDATE 1

postgres=*# commit;
COMMIT
postgres=#

```

Reply:
Checking [the docs](https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-ROWS) you're right 

(though one small thing, SERIALIZABLE will make concurrent TXs error if the record changed since they started acquiring their locks)

Should you like me to revert to the prev implementation?

Reply:
I was testing with SERIALIZABLE, but we should not use that for transaction coordination.